ducktape/tests/test.py

```diff
@@ -334,18 +336,55 @@ def results_dir(test_context, test_index):
 
     @property
     def expected_num_nodes(self):
-        """How many nodes we expect this test to consume when run.
+        """
+        How many nodes we expect this test to consume when run.
+
+        This default implementation assumes all nodes are Linux.
+
         Return None if undefined.
         """
+        node_spec = self.expected_node_spec
+        if node_spec is None:
+            return None
+
+        count = 0
+        for (_, node_count) in node_spec.iteritems():
+            count += node_count
+        return count
+
+    @property
+    def expected_node_spec(self):
+        """
+        How many nodes we expect for each operating system.
+
+        This default implementation assumes all nodes are Linux.
+
+        Return None if undefined.
+        """
+
+        # TODO: closely code review this, please. I can't find anything in ducktape or muckrake that
```

Comment:
The `expected_num_nodes` has limited use these days. We ended up going the annotation route eventually partly because variants of the same test can end up using a different # of nodes. We should probably even just get rid of it entirely, so adding to it with `expected_node_spec` may not make sense.